### PR TITLE
WT-10538 Remove redundant compare and swap

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/btree/row_modify.c [hunk 1]
```diff
@@ -392,8 +396,12 @@ __wt_update_obsolete_check(
      * subsequent to it, other threads of control will terminate their walk in this element. Save a
      * reference to the list we will discard, and terminate the list.
      */
-    if (first != NULL && (next = first->next) != NULL &&
-      __wt_atomic_cas_ptr(&first->next, next, NULL)) {
+    if (first != NULL && (next = first->next) != NULL) {
+        /*
+         * No need to use a compare and swap because we have obtained a lock before entering this
+         * function.
```
Comment (korteland) [inline src/btree/row_modify.c:402]: This isn't quite correct, we acquired the lock at the start of this function.

File: src/btree/row_modify.c [hunk 2]
```diff
@@ -392,8 +396,12 @@ __wt_update_obsolete_check(
...
+         */
+        first->next = NULL;
         /*
```
Comment (korteland) [inline src/btree/row_modify.c:405]: A blank line here would imo help readability. Feel free to disregard this if you don't agree.

**Review by korteland — COMMENTED** [review]
Code changes LGTM! Two tiny issues.

**Review by korteland — APPROVED** [review]
LGTM!

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Comment by quchenhao** [comment]
evergreen merge

**Pull request merged**